pkg/pillar/kubeapi/nodedrain.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build kubevirt
+
+package kubeapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+)
+
+// RequestNodeDrain generates the NodeDrainRequest object and publishes it
+func RequestNodeDrain(pubNodeDrainRequest pubsub.Publication, requester DrainRequester, context string) error {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("RequestNodeDrain: can't get hostname %v", err)
+	}
+	drainReq := NodeDrainRequest{
+		Hostname:    hostname,
+		RequestedAt: time.Now(),
+		RequestedBy: requester,
+		Context:     context,
+	}
+	err = pubNodeDrainRequest.Publish("global", drainReq)
+	if err != nil {
+		return fmt.Errorf("RequestNodeDrain: error publishing drain request: %v", err)
+	}
+	return nil
+}
+
+// GetDrainStatusOverride : an alternate way to set drain status for debug
+func GetDrainStatusOverride(log *base.LogObject) *NodeDrainStatus {
+	// An alternate path to force a drain status in the event of a drain issue.
+	forceNodeDrainPath := "/persist/kube-status/force-NodeDrainStatus-global.json"
```

Comment:
I don't see the code which writes this file in /persist, but in general we need to use WriteRename() for such files to avoid partial file content due to power outage etc.

Reply:
The initial creator/user of this force file was myself in dev/debug.  There actually isn't any path in eve which writes this file out.